3p/3P_CONTRIBUTIONS.md

```diff
@@ -0,0 +1,53 @@
+# Inclusion of third party software into AMP
+
+In general all inclusions are subject to [CONTRIBUTING.md](../CONTRIBUTING.md).
+
+This files outlines specific rules for certain types of external embed and other software inclusions.
+
+## Fonts
+
+- AMP allows inclusion of fonts via the `@font-face` directive.
+- JavaScript can not be involved with the initiation of font loading.
+- AMP by default does not allow inclusion of external stylesheets, but it is happy to whitelist URL prefixes of font providers for font inclusion via link tags. These link tags and their fonts must be served via HTTPS.
+- If a font provider does referrer based "security" it needs to whitelist the AMP proxy origins before being included in the link tag whitelist.
+
+## Iframe based embeds
+
+- Our intent is to provide first class support for all
+- Consider whether a click-to-play solution fits your use case where iframe generation is not done immediately (can be done before user action for instant loading of iframe).
+- Consider whether all that is needed is some documentation for how to use the embed with `amp-iframe`.
+- Iframes and all sub resources must be served from HTTPS.
+- Avoid client side rendering of iframe content.
+- If your use of iframes is for style isolation, consider that AMP might provide an iframe-free alternative.
+- If you can make it not iframe based that is much better. (See e.g. the pinterest embed). We will always ask to do this first. E.g. adding a CORS endpoint to your server might make this possible.
+- Must play well within AMP's sizing framework.
+- All JS on container page must be open source and bundled with AMP.
+- Direct iframe embeds not using our 3p iframe mechanism (used e.g. for ads) are preferred.
+- JavaScript loaded into iframe should be reasonable with respect to functionality.
+- Use the `sandbox` attribute on iframe if possible.
+- Embeds that require plugins such as Flash or Java applets need to go through special review.
```

Comment:
We should probably word a bit stronger: "Embeds that require browser plugins, such as Flash, Java, ActiveX, Silverlight, etc. are disallowed unless necessary"